### PR TITLE
[WIP] missing-lang redirect fix

### PR DESCRIPTION
**Pull request by ginold**
so i've found a solution, but there is clearly something wrong that i can't achieve.
It seems that redirection works only from a directive (I remember it worked with edit-button, then it stopped). I tried every way around but this is the only one that works.
#274

### CODE REVIEW

**Comment by asaunier** [comment]
Instead of creating a new ``missingLang``directive (that duplicates the ``redirectToProtectedUrl`` method from ``mainController``), what about re-adding the ``editButton`` directive and use it everywhere we need to check the user credentials before granting him access:
https://github.com/c2corg/v6_ui/search?utf8=%E2%9C%93&q=redirectToProtectedUrl

**Comment by ginold** [comment]
i hardly believe that what we want to do is impossible and that we have to create a directive specially for that! can someone have a look and try ?

**Comment by asaunier** [comment]
I would simply get the ``editButton`` directive back:
https://github.com/c2corg/v6_ui/blob/d228cfc4a324873ad5cb88299fe119b44e15e068/c2corg_ui/static/js/editbutton.js

I don't remember exactly but I think I did it this way because it failed the other way. In addition the ``editButton`` directive add a transclusion that allows to easily embed anything inside the button:
https://github.com/c2corg/v6_ui/blob/d228cfc4a324873ad5cb88299fe119b44e15e068/c2corg_ui/static/js/editbutton.js#L25

Renaming the directive is of course an option if we want to use it for anything else than editing actions.

**Comment by ginold** [comment]
maybe simply call it redirect-button ?

**Comment by asaunier** [comment]
As you want provided it is not too long, self-explicity and generic enough.
What about simply ``protected-button``?

**Comment by ginold** [comment]
i chose protected-url-btn.
 Have a look!

File: c2corg_ui/static/js/protectedurlbtn.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+goog.provide('app.protectedUrlBtnController');
```
Comment (asaunier) [inline c2corg_ui/static/js/protectedurlbtn.js]: ``protectedUrlBtnController`` => ``ProtectedUrlBtnController`` (capital initial letter for controllers)

**Comment by asaunier** [comment]
I have not tested but the code looks good.

**Comment by ginold** [comment]
because it is good :+1:  I will update and merge, ok?

**Pull request merged**